File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    let sum = 0;
    
    for(let i = left; i <= right; i++){
        const count = divisorCounter(i);
        sum += count % 2 === 0 ? i : i * -1
    }
   
    return sum;
    
}

const divisorCounter = (num) => {
    // 약수 배열
    let count = 0;
    
    for(let i = 1; i <= Math.floor(Math.sqrt(num)); i++){
        
        if(num % i === 0) {
            count++;
            
            if(i !== num / i) count++;
            
        }
    }
    
    return count;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
import math

def solution(left, right):
    
    sum = 0
    
    for num in range(left, right + 1):
        
        if((numOfDivisors(num) % 2) == 0):
            sum += num
        else:
            sum -= num
        
    return sum
    
def numOfDivisors(num):
        count = 0
        
        sqrt = math.isqrt(num)
        
        for i in range(1, sqrt + 1):
            
            if(num % i) == 0:
                if((num // i) != sqrt):
                    count += 2
                else:
                    count += 1
                    
        return count